voice-assistant/esp32-s3-box-3.yaml

```
@@ -149,65 +159,77 @@ voice_assistant:
   on_listening:
     - display.page.show: listening_page
     - component.update: s3_box_lcd
-  on_tts_start:
-    - display.page.show: talking_page
-    - component.update: s3_box_lcd
-  on_end:
-    - delay: 100ms
-    - wait_until:
-        not:
-          speaker.is_playing:
-    - script.execute: reset_display
   on_stt_vad_end:
     - display.page.show: thinking_page
     - component.update: s3_box_lcd
-  on_error:
-    - display.page.show: error_page
+  on_tts_stream_start:
+    - display.page.show: replying_page
     - component.update: s3_box_lcd
-    - delay: 1s
+  on_tts_stream_end:
     - script.execute: reset_display
+  on_error:
+    - if:
+        condition:
+          lambda: return !id(init_in_progress);
+        then:  
+          - display.page.show: error_page
+          - component.update: s3_box_lcd
+          - delay: 1s
+          - script.execute: reset_display
   on_client_connected:
     - if:
         condition:
           switch.is_on: use_wake_word
         then:
           - voice_assistant.start_continuous:
```

A small thing I found if API/HA connected before the above 5 second delay. The mic would not start due to ram still used by BLE.
```suggestion
          - wait_until:
              not: ble.enabled
          - voice_assistant.start_continuous:
```